pkg/util/k8sutil/k8sutil.go

```diff
@@ -131,14 +143,28 @@ func ClientServiceName(clusterName string) string {
 	return clusterName + "-client"
 }
 
-func CreatePeerService(kubecli kubernetes.Interface, clusterName, ns string, owner metav1.OwnerReference) error {
+func CreatePeerService(kubecli kubernetes.Interface, clusterName, ns string, isSecureClient, isSecurePeer bool, owner metav1.OwnerReference) error {
```

Comment:
Can you change this func to accept `clientPortName`, `serverPortName` instead?
This util shouldn't understand etcd logic.

Reply:
Done, although if these utilities shouldn't understand etcd logic at all I'd suggest removing the hardcoded port numbers (on a separate PR, perhaps).